extensions/ql-vscode/src/query-history/query-history-manager.ts

```diff
@@ -627,36 +610,32 @@ export class QueryHistoryManager extends DisposableObject {
   }
 
   async handleItemClicked(
-    singleItem: QueryHistoryInfo | undefined,
+    singleItem: QueryHistoryInfo,
     multiSelect: QueryHistoryInfo[] | undefined,
   ) {
-    const { finalSingleItem, finalMultiSelect } = this.determineSelection(
-      singleItem,
-      multiSelect,
-    );
-    if (!this.assertSingleQuery(finalMultiSelect) || !finalSingleItem) {
+    if (!this.assertSingleQuery(multiSelect)) {
```

Comment:
Not a problem with your PR but "assert" makes me think an error will be thrown if the expectation isn't met. Perhaps we should rename to `isSingleQuery` or something like that in the future.

Reply:
Yes I agree it's confusingly named. I actually plan to delete it entirely in https://github.com/github/vscode-codeql/pull/2287